### PR TITLE
feature(advanced-rpc-compression): new test configuration

### PR DESCRIPTION
**Pull request by dimakr**
A new feature in scylladb allows to enable advanced compression of internode rpc messaging.

The change adds the following to enable the feature in a scylladb cluster under test:
- new attributes that enable advanced compression in scylla are added to ScyllaYaml model (with empty values by default)
- a new test configuration is added, to be used for enabling advanced compression by extending existing test configurations

It also enables the feature in 'rolling-upgrade-multidc' and tier1 'longevity-50gb-3days' jenkins jobs.

### Testing

- [x] :green_circle: [generic longevity-5gb-1h-nemesis test with the new 'configurations/advanced-rpc-compression.yaml' configuration included](https://argus.scylladb.com/test/8113a9b7-6fe7-4ad4-9663-eda51b2938a1/runs?additionalRuns[]=6d82bc78-379b-4f52-aaf9-cde62183d382)

### PR pre-checks (self review)
- [ ] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by dimakr** [comment]
@fruch

As we discussed - I checked enabling the advanced rpc compression scylla.yaml parameters in scylla on oss, where the feature is not available.
Scylla is OK with having these parameters, just warns in logs that there some unknown scylla options, e.g.
```
db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/scylla.yaml:rpc_dict_training_min_bytes: 100000000
db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/scylla.yaml:rpc_dict_training_min_time_seconds: 900
db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/scylla.yaml:rpc_dict_training_when: when_leader
db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/scylla.yaml:rpc_dict_update_period_seconds: 150
...
db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/messages.log:2024-07-10T10:10:40.664+00:00 ubuntu-advanced-db-node-6d82bc78-1  !WARNING | scylla[6631]:  [shard 0:main] init - Unknown option : rpc_dict_training_min_bytes
db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/messages.log:2024-07-10T10:10:40.664+00:00 ubuntu-advanced-db-node-6d82bc78-1  !WARNING | scylla[6631]:  [shard 0:main] init - Unknown option : rpc_dict_training_min_time_seconds
db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/messages.log:2024-07-10T10:10:40.664+00:00 ubuntu-advanced-db-node-6d82bc78-1  !WARNING | scylla[6631]:  [shard 0:main] init - Unknown option : rpc_dict_training_when
db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/messages.log:2024-07-10T10:10:40.664+00:00 ubuntu-advanced-db-node-6d82bc78-1  !WARNING | scylla[6631]:  [shard 0:main] init - Unknown option : rpc_dict_update_period_seconds
```

So should we enable the feature in few generic jobs that are running regularly on master, but are also picked up for release testing?
As per results above it makes no harm and adds just a bit of one time noise per a node in logs, when scylla is started.

Or should we enable it in release jobs only?

**Review by fruch — DISMISSED** [review]

**Comment by fruch** [comment]
> @fruch
> 
> As we discussed - I checked enabling the advanced rpc compression scylla.yaml parameters in scylla on oss, where the feature is not available.
> Scylla is OK with having these parameters, just warns in logs that there are some unknown scylla options, e.g.
> ```
> db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/scylla.yaml:rpc_dict_training_min_bytes: 100000000
> db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/scylla.yaml:rpc_dict_training_min_time_seconds: 900
> db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/scylla.yaml:rpc_dict_training_when: when_leader
> db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/scylla.yaml:rpc_dict_update_period_seconds: 150
> ...
> db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/messages.log:2024-07-10T10:10:40.664+00:00 ubuntu-advanced-db-node-6d82bc78-1  !WARNING | scylla[6631]:  [shard 0:main] init - Unknown option : rpc_dict_training_min_bytes
> db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/messages.log:2024-07-10T10:10:40.664+00:00 ubuntu-advanced-db-node-6d82bc78-1  !WARNING | scylla[6631]:  [shard 0:main] init - Unknown option : rpc_dict_training_min_time_seconds
> db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/messages.log:2024-07-10T10:10:40.664+00:00 ubuntu-advanced-db-node-6d82bc78-1  !WARNING | scylla[6631]:  [shard 0:main] init - Unknown option : rpc_dict_training_when
> db-cluster-6d82bc78/ubuntu-advanced-db-node-6d82bc78-1/messages.log:2024-07-10T10:10:40.664+00:00 ubuntu-advanced-db-node-6d82bc78-1  !WARNING | scylla[6631]:  [shard 0:main] init - Unknown option : rpc_dict_update_period_seconds
> ```
> 
> So should we enable the feature in few generic jobs that are running regularly on master, but are also picked up for release testing?
> As per results above it makes no harm and adds just a bit of one time noise per a node in logs, when scylla is started.
> 
> Or should we enable it in release jobs only?

If it's just a warning in the log, we can enable it on OSS jobs as well

we should pick the on relevert tier1 and add it to it

And backport it to the enterprise release branch it's needed in (2024.2 I'm guessing)

**Review by soyacz — APPROVED** [review]
LGTM
Btw. would be nice to add link to doc's about this feature in config yaml so we can understand the parameters easier/faster.

**Comment by fruch** [comment]
> LGTM Btw. would be nice to add link to doc's about this feature in config yaml so we can understand the parameters easier/faster.

if only each flag there had docs :)

**Comment by soyacz** [comment]
> > LGTM Btw. would be nice to add link to doc's about this feature in config yaml so we can understand the parameters easier/faster.
> 
> if only each flag there had docs :)

raise doc's issue if not. What's the point of having configuration that no one understands?

**Comment by dimakr** [comment]
> we should pick the on relevert tier1 and add it to it

@fruch 
Added test configuration for 'rolling-upgrade-multidc.jenkinsfile' and 'longevity-50gb-3days' jobs.

File: jenkins-pipelines/oss/rolling-upgrade/rolling-upgrade-multidc.jenkinsfile [hunk 1]
```diff
@@ -10,6 +10,6 @@ rollingUpgradePipeline(
     linux_distro: 'ubuntu-focal',
 
     test_name: 'upgrade_test.UpgradeTest.test_generic_cluster_upgrade',
-    test_config: 'test-cases/upgrades/multi-dc-rolling-upgrade.yaml',
+    test_config: '''["test-cases/upgrades/multi-dc-rolling-upgrade.yaml", "configurations/advanced-rpc-compression.yaml"]'''
```
Comment (fruch) [inline jenkins-pipelines/oss/rolling-upgrade/rolling-upgrade-multidc.jenkinsfile:13]: you missing a comma at the end of this line

Comment (dimakr) [reply jenkins-pipelines/oss/rolling-upgrade/rolling-upgrade-multidc.jenkinsfile]: added

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**